packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1811,6 +1816,111 @@ export function catalogServiceBuilder(
       return updatedEservice;
     },
+    async updateEServiceFlags(
+      eserviceId: EServiceId,
+      {
+        isDelegable,
+        isClientAccessDelegable,
+      }: {
+        isDelegable: boolean;
+        isClientAccessDelegable: boolean;
+      },
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<EService> {
+      logger.info(`Updating EService ${eserviceId} flags`);
+      const eservice = await retrieveEService(eserviceId, readModelService);
+
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        authData,
+        readModelService
+      );
+
+      const hasValidDescriptor = eservice.data.descriptors.some(
+        // eslint-disable-next-line sonarjs/no-identical-functions
+        (descriptor) =>
+          descriptor.state !== descriptorState.draft &&
+          descriptor.state !== descriptorState.waitingForApproval &&
+          descriptor.state !== descriptorState.archived
+      );
+      if (!hasValidDescriptor) {
+        throw eserviceWithoutValidDescriptors(eserviceId);
+      }
+
+      if (!isDelegable && isClientAccessDelegable) {
+        throw invalidEServiceFlags(eserviceId);
+      }
+
+      const updatedEservice: EService = {
+        ...eservice.data,
+        isDelegable,
+        isClientAccessDelegable,
+      };
+
+      const event = match({
+        isDelegable,
+        oldIsDelegable: eservice.data.isDelegable || false,
+        isClientAccessDelegable,
+        oldIsClientAccessDelegable:
+          eservice.data.isClientAccessDelegable || false,
+      })
+        .with(
+          {
+            isDelegable: true,
+            oldIsDelegable: false,
+          },
+          () =>
+            toCreateEventEServiceIsDelegableEnabled(
+              eservice.metadata.version,
+              updatedEservice,
+              correlationId
+            )
+        )
+        .with(
+          {
+            isDelegable: false,
+            oldIsDelegable: true,
+          },
+          () =>
+            toCreateEventEServiceIsDelegableDisabled(
+              eservice.metadata.version,
+              updatedEservice,
+              correlationId
+            )
+        )
+        .with(
+          {
+            isClientAccessDelegable: true,
+            oldIsClientAccessDelegable: false,
+          },
+          () =>
+            toCreateEventEServiceIsClientAccessDelegableEnabled(
+              eservice.metadata.version,
+              updatedEservice,
+              correlationId
+            )
+        )
+        .with(
+          {
+            isClientAccessDelegable: false,
+            oldIsClientAccessDelegable: true,
+          },
+          () =>
+            toCreateEventEServiceIsClientAccessDelegableDisabled(
+              eservice.metadata.version,
+              updatedEservice,
+              correlationId
+            )
+        )
+        .otherwise(() => undefined);
```

Comment:
That's true and I already think about it. There are three possible solutions:
- let the user update one flag for each call
- this one where we can update both flags (usually the consumer overrides the current read-model with the one in the event) but we lost an event. I mean we are saying that the `IsDelegableEnabled` event could also activate the `isClientAccessDelegable` flag
- produce multiple events with a single call. It's something that we have done in the past but since the V2 events it's  deprecated

Reply:
That's true and I already think about it. There are three possible solutions:
- let the user update one flag for each call
- this one where we can update both flags (usually the consumer overrides the current read-model with the one in the event) but we lost an event. I mean we are saying that the `IsDelegableEnabled` event could also activate the `isClientAccessDelegable` flag
- produce multiple events with a single call. It's something that we have done in the past but since the V2 events it's  deprecated (at least for the same streamId)

Reply:
That change was introduced after the eventsV2 migration and it isn't so nice. We are creating 2 events for the same service (the first one containing all the eservice data without the descriptor and the second one the data present in the first one + the descriptor). It also works only for an edge case of the createEvents function because both events has as version number 0.

I presume that the only purpose was to have the `DescriptionAdd` event type. In that case, a better approach would be to add a new event like "CreateEServiceWithDescriptor" or something like that.

We have made the V1 to V2 event to have all the possible entity data inside the event and to avoid this kind of situation so it sounds strange that we are on that path again

Reply:
about the exhaustive it's a pretty big list if I want to declare all the other possible cases. I will try to find another way